std-bits/table/src/main/java/org/enso/table/excel/xssfreader/XSSFReaderSheet.java

```diff
@@ -0,0 +1,163 @@
+      var styles = parent.getStyles();
+      var handler =
+          new XSSFReaderSheetXMLHandler(styles, strings) {
+            @Override
+            protected void onDimensions(String dimension) {
+              handleOnDimensions(dimension);
+            }
+
+            @Override
+            protected void onStartRow(int rowNum) {
+              handleOnStartRow(rowNum);
+            }
+
+            @Override
+            protected void onCell(int rowNumber, short columnNumber, String ref, CellValue value) {
+              handleOnCell(rowNumber, columnNumber, value);
+            }
+          };
+
+      var xmlReader = XMLHelper.newXMLReader();
+      xmlReader.setContentHandler(handler);
+
+      rowData = new HashMap<>();
+
+      try {
+        parent.withReader(
+            reader -> {
+              try {
+                var sheet = reader.getSheet(relId);
+                xmlReader.parse(new InputSource(sheet));
+              } catch (SAXException | InvalidFormatException | IOException e) {
+                throw new RuntimeException(e);
+              }
+            });
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+
+      readSheetData = true;
+    } catch (SAXException | ParserConfigurationException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public int getSheetIndex() {
+    return sheetIdx;
+  }
+
+  @Override
+  public String getName() {
+    return sheetName;
+  }
+
+  public String getDimensions() {
+    if (!readSheetData) {
+      readSheetData();
+    }
+    return dimensions;
+  }
+
+  @Override
+  public int getFirstRow() {
+    if (!readSheetData) {
+      readSheetData();
+    }
+    return firstRow;
+  }
+
+  @Override
+  public int getLastRow() {
+    if (!readSheetData) {
+      readSheetData();
+    }
+    return lastRow;
+  }
+
+  @Override
+  public ExcelRow get(int row) {
+    if (!readSheetData) {
+      readSheetData();
+    }
+
+    if (!rowData.containsKey(row)) {
+      return null;
+    }
+
+    return new XSSFReaderRow(rowData.get(row), parent.use1904Format());
+  }
+
+  @Override
+  public Sheet getSheet() {
+    // Not supported as we don't have the underlying Apache POI Sheet object.
+    return null;
+  }
+
+  protected void handleOnDimensions(String dimension) {
+    dimensions = dimension;
+  }
+
+  private void handleOnStartRow(int rowNum) {
+    if (firstRow == 0 || rowNum < firstRow) {
+      firstRow = rowNum;
+    }
+
+    if (lastRow == 0 || rowNum > lastRow) {
+      lastRow = rowNum;
+    }
+  }
+
+  private void handleOnCell(
+      int rowNumber, short columnNumber, XSSFReaderSheetXMLHandler.CellValue value) {
+    if (rowData.containsKey(rowNumber)) {
+      rowData.get(rowNumber).put(columnNumber, value);
+    } else {
+      var map = new TreeMap<Short, XSSFReaderSheetXMLHandler.CellValue>();
+      map.put(columnNumber, value);
+      rowData.put(rowNumber, map);
+    }
```

Comment:
nitpick, but this is usually handled by the `computeIfAbsent` idiom. I think using it will make the code easier to understand as one can just spot the `computeIfAbsent` pattern VS having to analyze what the branches of this `if` statement do.

```suggestion
    rowData.computeIfAbsent(rowNumber, k -> new TreeMap<Short, XSSFReaderSheetXMLHandler.CellValue>())
      .put(columnNumber, value);
```